hack/ci/jobguard.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e
+
+JOB_NAME_PATTERN=${JOB_NAME_PATTERN:-"(pre-main-kyma-components-.*)|(pre-main-kyma-tests-.*)|(pre-kyma-components-.*)|(pre-kyma-tests-.*)|(pull-.*-build)"}
```

Comment:
Do we need the patterns "pre-main-kyma-components-.*", "pre-main-kyma-tests-.*", "pre-kyma-components-.*" and "pre-kyma-tests-.*"? As far as I understand we only use this as PR build guard for now and therefore "pull-.*-build" should be enough to avoid confusion.

Reply:
Your right, I left that for consistency with API Gateway, but we should probably just remove it
```suggestion
JOB_NAME_PATTERN=${JOB_NAME_PATTERN:-"(pull-.*-build)"}
```